Source/santametricservice/Writers/SNTMetricHTTPWriter.m

```diff
@@ -0,0 +1,101 @@
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///    http://www.apache.org/licenses/LICENSE-2.0
+///
+///    Unless required by applicable law or agreed to in writing, software
+///    distributed under the License is distributed on an "AS IS" BASIS,
+///    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+///    See the License for the specific language governing permissions and
+///    limitations under the License.
+#import <MOLAuthenticatingURLSession/MOLAuthenticatingURLSession.h>
+#include <dispatch/dispatch.h>
+
+#import "Source/common/SNTLogging.h"
+#import "Source/santametricservice/Writers/SNTMetricHTTPWriter.h"
+
+@implementation SNTMetricHTTPWriter {
+ @private
+  MOLAuthenticatingURLSession *_authSession;
+}
+
+- (instancetype)init {
+  self = [super init];
+  if (self) {
+    _authSession = [[MOLAuthenticatingURLSession alloc] init];
+  }
+  return self;
+}
+
+/**
+ * Post serialzied metrics to the specified URL one object at a time.
+ **/
+- (BOOL)write:(NSArray<NSData *> *)metrics toURL:(NSURL *)url error:(NSError **)error {
+  __block NSError *_blockError = nil;
+
+  NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:url];
+  request.HTTPMethod = @"POST";
+  [request setValue:@"application/json" forHTTPHeaderField:@"Content-Type"];
+
+  _authSession.serverHostname = url.host;
+  NSURLSession *_session = _authSession.session;
+
+  dispatch_group_t requests = dispatch_group_create();
+
+  [metrics enumerateObjectsUsingBlock:^(id value, NSUInteger index, BOOL *stop) {
+    dispatch_group_enter(requests);
+
+    request.HTTPBody = (NSData *)value;
+    [[_session dataTaskWithRequest:request
+                 completionHandler:^(NSData *_Nullable data, NSURLResponse *_Nullable response,
+                                     NSError *_Nullable err) {
+                   if (err != nil) {
+                     _blockError = err;
+                     *stop = YES;
+                   } else if (response == nil) {
+                     *stop = YES;
+                   } else if ([response isKindOfClass:[NSHTTPURLResponse class]]) {
+                     NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse *)response;
+
+                     // Check HTTP error codes and create errors for any non-200.
+                     if (httpResponse && httpResponse.statusCode != 200) {
+                       _blockError = [[NSError alloc]
+                         initWithDomain:@"com.google.santa.metricservice.writers.http"
+                                   code:httpResponse.statusCode
+                               userInfo:@{
+                                 NSLocalizedDescriptionKey : [NSString
+                                   stringWithFormat:@"received http status code %ld from %@",
+                                                    httpResponse.statusCode, url]
+                               }];
```

Comment:
Did clang-format format this?

Reply:
Yes.

Reply:
Not asking you to change it, but this is almost unreadable 😞